lib/media/track/mediatrack.js

```diff
@@ -195,9 +197,9 @@ class MediaTrack extends Track {
       mediaStream.addTrack(mediaStreamTrack);
     }
 
-    // NOTE(mpatwardhan): resetting `srcObject` here, causes flicker (JSDK-2641), but it lets us
-    // sidestep the chrome bug: https://bugs.chromium.org/p/chromium/issues/detail?id=1052353
-    el.srcObject = mediaStream;
+    if (!el.srcObject) {
+      el.srcObject = mediaStream;
+    }
```

Comment:
Is there a situation where we have to update the srcObject after it's already set?

Reply:
yes, for dummyElement now _attach gets called everytime track switches. But in that case it woudl already have a srcObject from previous track.